nucleus/common/jul-extensions/src/main/java/fish/payara/logging/jul/cfg/SortedProperties.java

```diff
@@ -0,0 +1,204 @@
+ *  The Payara Foundation designates this particular file as subject to the "Classpath"
+ *  exception as provided by the Payara Foundation in the GPL Version 2 section of the License
+ *  file that accompanied this code.
+ *
+ *  Modifications:
+ *  If applicable, add the following below the License Header, with the fields
+ *  enclosed by brackets [] replaced by your own identifying information:
+ *  "Portions Copyright [year] [name of copyright owner]"
+ *
+ *  Contributor(s):
+ *  If you wish your version of this file to be governed by only the CDDL or
+ *  only the GPL Version 2, indicate your decision by adding "[Contributor]
+ *  elects to include this software in this distribution under the [CDDL or GPL
+ *  Version 2] license."  If you don't indicate a single choice of license, a
+ *  recipient has the option to distribute your version of this file under
+ *  either the CDDL, the GPL Version 2 or to extend the choice of license to
+ *  its licensees as provided above.  However, if you add GPL Version 2 code
+ *  and therefore, elected the GPL Version 2 license, then the option applies
+ *  only if the new code is made subject to such option by the copyright
+ *  holder.
+ */
+
+package fish.payara.logging.jul.cfg;
+
+import java.io.BufferedOutputStream;
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.Enumeration;
+import java.util.LinkedHashSet;
+import java.util.Map;
+import java.util.Properties;
+import java.util.Set;
+import java.util.SortedSet;
+import java.util.TreeSet;
+import java.util.stream.Collectors;
+
+/**
+ * Extended {@link Properties} class. Same as {@link Properties}, it uses hashcodes for providing
+ * values for property names, but the difference is in storing content to a file or stream, which is
+ * sorted by keys.
+ * <p>
+ * Also provides some methods for more comfortable work.
+ * <p>
+ * Warning: using other than {@link String} objects can cause unexpected behavior.
+ *
+ * @author David Matejcek
+ */
+// note: this class must not use anything from JUL or PJULE packages, because the dependency is exactly opposite.
+public class SortedProperties extends Properties {
+
+    private static final long serialVersionUID = -2007016845217920652L;
+
+    /**
+     * Creates an empty property table.
+     */
+    public SortedProperties() {
+        super();
+    }
+
+
+    /**
+     * Makes a copy of provided properties.
+     *
+     * @param properties
+     */
+    public SortedProperties(final Properties properties) {
+        super();
+        putAll(properties);
+    }
+
+
+    /**
+     * @return a sorted {@link Enumeration} of keys.
+     */
+    // affects storage!
+    @Override
+    public Enumeration<Object> keys() {
```

Comment:
Nope. Evil is hidden in java.util.Properties and until Java would get rid of it, it would be more work to replace it with all it's good and bad features.